src/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobReader.cs

```diff
@@ -87,55 +86,43 @@ internal string GetDebuggerDisplay()
         /// <summary>
         /// The total length of the underlying memory block.
         /// </summary>
-        public int Length => _block.Length;
+        public int Length => (int)(_endPointer - _startPointer);
 
         /// <summary>
         /// Offset from start of underlying memory block to current position.
         /// </summary>
-        public int Offset => (int)(_currentPointer - _block.Pointer);
-
-        /// <summary>
-        /// Bytes remaining from current position to end of underlying memory block.
-        /// </summary>
-        public int RemainingBytes => (int)(_endPointer - _currentPointer);
-       
-        /// <summary>
-        /// Repositions the reader to the start of the underluing memory block.
-        /// </summary>
-        public void Reset()
-        {
-            _currentPointer = _block.Pointer;
-        }
-
-        /// <summary>
-        /// Repositions the reader to the given offset from the start of the underlying memory block.
-        /// </summary>
-        /// <exception cref="BadImageFormatException">Offset is outside the bounds of underlying reader.</exception>
-        public void SeekOffset(int offset)
+        /// <exception cref="BadImageFormatException">Offset is set outside the bounds of underlying reader.</exception>
+        public int Offset
         {
-            if (!TrySeekOffset(offset))
+            get
             {
-                Throw.OutOfBounds();
+                return (int)(_currentPointer - StartPointer);
             }
-        }
 
-        internal bool TrySeekOffset(int offset)
-        {
-            if (unchecked((uint)offset) >= (uint)_block.Length)
+            set
             {
-                return false;
-            }
+                byte* newPointer = StartPointer + value;
 
-            _currentPointer = _block.Pointer + offset;
-            return true;
+                if (newPointer > _endPointer)
+                {
+                    Throw.OutOfBounds();
+                }
+
+                _currentPointer = newPointer;
```

Comment:
Could this just be:
```C#
_currentPointer = GetPointerChecked(value);
```
?

(Or, I may just be confused by the diff, and that GetPointerChecked is actually on a different type.)